src/test/java/com/flightstats/hub/webhook/WebhookLeaderLocksTest.java

```diff
@@ -46,20 +44,14 @@ void deletePath() throws Exception {
         curator.delete().deletingChildrenIfNeeded().forPath(WEBHOOK_LEADER_PATH);
     }
 
-    @BeforeAll
-    static void setup() {
-        curator = IntegrationTestSetup.run().getZookeeperClient();
-        zooKeeperUtils = new SafeZooKeeperUtils(curator);
-    }
-
     @BeforeEach
     void createWebhookLeader() throws Exception {
+        curator = IntegrationTestSetup.run().getZookeeperClient();
+        zooKeeperUtils = new SafeZooKeeperUtils(curator);
         try {
             createPath();
         } catch (NodeExistsException e) {
-            log.error(e.getMessage());
-            deletePath();
-            createPath();
+            log.error("webhook leader path already exists");
```

Comment:
Deleting and recreating the path was causing issues? I would have thought that clearing out the state would help with state bleed-over.

Reply:
It was not, no, it is not necessary though, moving the curator object creation into the before each instead of the before all isolated the state and allowed the delete path in the after each to work as intended